src/internal/connector/onedrive/restore.go

```diff
@@ -65,18 +68,20 @@ func RestoreCollections(
 
 	// Iterate through the data collections and restore the contents of each
 	for _, dc := range dcs {
-		parentPerms, ok := parentPermissions[dc.FullPath().String()]
-		if !ok {
-			onedrivePath, err := path.ToOneDrivePath(dc.FullPath())
-			if err != nil {
-				errUpdater(dc.FullPath().String(), fmt.Errorf("invalid restore path"))
-			}
+		if opts.RestorePermissions {
```

Comment:
Nit: we're getting some pretty deeply nested code here.  I think this is a great opportunity to break it out (lines 72-83) into a separate function, both for cleanliness and ease of unit testing.
```go
if opts.RestorePermissions {
  if err := restorePermssion(dc.FullPath()); err != nil {
    errUpdater(err)
  }
}
```